src/Microsoft.AspNetCore.Diagnostics/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -126,42 +126,66 @@ private Task DisplayException(HttpContext context, Exception ex)
                 Options = _options,
             };
 
+            var errorPage = new CompilationErrorPage
+            {
+                Model = model
+            };
+
+            if (compilationException.CompilationFailures == null)
+            {
+                return errorPage.ExecuteAsync(context);
+            }
+
             foreach (var compilationFailure in compilationException.CompilationFailures)
```

Comment:
Could we refactor this in a different method so we don't have two different calls to `ExexcuteAsync`. ie

```
var errorPage = new CompilationErrorPage
{
   Model = model
};

ReadCompilationErrors(model);
return errorPage.ExecuteAsync(context);
```